sporkfish/searcher/minimax.py

```diff
@@ -0,0 +1,253 @@
+import copy
+import logging
+import os
+import time
+from abc import ABC, abstractmethod
+from typing import Any, Callable, List, Optional, Tuple
+
+import chess
+import stopit
+from pathos.multiprocessing import ProcessPool
+
+from ..board.board import Board
+from ..evaluator import Evaluator
+from ..statistics import Statistics
+from ..transposition_table import TranspositionTable
+from ..zobrist_hasher import ZobristHasher
+from .move_ordering import MoveOrder, MvvLvaHeuristic
+from .searcher import Searcher
+from .searcher_config import MoveOrdering, SearcherConfig, SearchMode
+
+_dict: dict = dict()
+
+
+class MiniMaxVariants(Searcher):
+    """
+    Abstract base class for minimax like searchers
+    """
+
+    @abstractmethod
+    def _searcher(
```

Comment:
Discussed offline, suggest a name like `start_search` to indicate starting the search from root.

Reply:
Ty for suggestion - amended